src/components/components/map/map.js

```diff
@@ -2,20 +2,49 @@ import './map.scss';
 
 import React from 'react';
 import { Map, Marker, Popup, TileLayer } from 'react-leaflet';
-
 import { ClubTeaser } from './../club-teaser-list/club-teaser-list';
 
 // local modules
 // styles
 class ClubMap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      clubs: this.props.clubs || [],
+      bounds: [[25.3345123, -99.5218668], [38.3345123, -80.5218668]]
+    };
+  }
+
+  componentDidMount = () => {
+    this.props.clubs !== this.state.clubs && this.updateClubs(this.props.clubs);
+  };
+
+  componentDidUpdate() {
+    this.props.clubs !== this.state.clubs && this.updateClubs(this.props.clubs);
+  }
+
+  updateClubs = clubs => {
+    let boundsArray = [];
+    this.props.clubs.map((club, i) => {
+      club.location && club.location.lat && club.location.lng
+        ? boundsArray.push([club.location.lat, club.location.lng])
+        : null;
+    });
+
+    // Ensure valid bounds with no or one result.
+    boundsArray.length < 2 && boundsArray.push(this.state.bounds);
+    this.setState({ clubs: this.props.clubs, bounds: boundsArray });
+  };
+
   render() {
-    // center the map.
-    const position = [38.3345123, -99.5218668];
-    const clubs = this.props.clubs || [];
+    const snapshot = this.state;
```

Comment:
This is an extra variable that isn't needed. 

Suggestion: Either use the full `this.state` or do `const { bounds, clubs } = this.state` and use the 2 variable afterwords.